### PR TITLE
Change precedence of assumption classes

### LINKED ISSUE #1961: Assumptions.assumeThat incompatible with TestNG 7.1.1+
See https://github.com/cbeust/testng/issues/2352

### PR DESCRIPTION
**Pull request by scordio**
Related to cbeust/testng#2352.

#### Check List:
* Fixes #1961 
* Unit tests : NA
* Javadoc with a code example (on API only) : NA

### CODE REVIEW

File: src/test/java/org/assertj/core/api/assumptions/Assumptions_assumeThat_Atomics_Test.java [hunk 1]
```diff
@@ -17,178 +17,167 @@
...
-import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
-import java.util.concurrent.atomic.AtomicLong;
-import java.util.concurrent.atomic.AtomicLongArray;
-import java.util.concurrent.atomic.AtomicLongFieldUpdater;
-import java.util.concurrent.atomic.AtomicMarkableReference;
-import java.util.concurrent.atomic.AtomicReference;
-import java.util.concurrent.atomic.AtomicReferenceArray;
-import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
-import java.util.concurrent.atomic.AtomicStampedReference;
+import java.util.concurrent.atomic.*;
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/assumptions/Assumptions_assumeThat_Atomics_Test.java:20]: detail but I prefer having all imports.

File: src/test/java/org/assertj/core/api/assumptions/Assumptions_assumeThat_Atomics_Test.java [hunk 2]
```diff
@@ -17,178 +17,167 @@
...
+                         assumeThat(actual).hasStamp(1);
+                       }
+                     });
   }
 
   @ParameterizedTest
   @MethodSource("provideAssumptionsRunners")
   void should_ignore_test_when_assumption_fails(AssumptionRunner<?> assumptionRunner) {
-    assertThatExceptionOfType(AssumptionViolatedException.class).isThrownBy(() -> assumptionRunner.runFailingAssumption());
+    assertThatExceptionOfType(TestAbortedException.class).isThrownBy(assumptionRunner::runFailingAssumption);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/assumptions/Assumptions_assumeThat_Atomics_Test.java:174]: let's replace all `assertThatExceptionOfType(TestAbortedException.class).isThrownBy` by the `expectAssumptionNotMetException` construct.

**Review by joel-costigliola — COMMENTED** [review]
thanks @scordio, looks good, just some minor comments

File: src/test/java/org/assertj/core/api/assumptions/Assumptions_assumeThat_Atomics_Test.java (hunk 1 above)
Comment (scordio) [reply src/test/java/org/assertj/core/api/assumptions/Assumptions_assumeThat_Atomics_Test.java:20]: Same here, I lost some configs during my last laptop change 🙈

**Comment by scordio** [comment]
@joel-costigliola PR ready

**Pull request merged**